src/endless/EndlessUsbToolDlg.cpp

```diff
@@ -2277,7 +2280,8 @@ void CEndlessUsbToolDlg::UpdateFileEntries(bool shouldInit)
             CString displayName, personality, version, date;
             bool isInstallerImage = false;
             if (!ParseImgFileName(trueName, personality, version, date, isInstallerImage)) continue;
-            ULONGLONG extractedSize = GetExtractedSize(currentFilePath, isInstallerImage);
+            CompressionType compressionType;
+            ULONGLONG extractedSize = GetExtractedSize(currentFilePath, isInstallerImage, compressionType);
```

Comment:
This confused me for quite a while because I couldn't see where compressionType was initialised any more - I guess C++ natives might be more used to it but for me the optimisation wasn't worth the readability hit. Maybe I just lament the fact you can't return a nice tuple... I don't have any particularly better suggestion - this function is a bit of a wasteground anyway. Making the files into classes rather than structs, might be a nice 2020 project! :)